<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Morgan Feeney, Leighton Pritchard" />


<title>Links to Key Public Data Resources</title>

<script src="database_list_files/header-attrs-2.10/header-attrs.js"></script>
<script src="database_list_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="database_list_files/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="database_list_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="database_list_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="database_list_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="database_list_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="database_list_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="database_list_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="database_list_files/navigation-1.1/tabsets.js"></script>
<link href="database_list_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="database_list_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="css/rmd_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">



<h1 class="title toc-ignore">Links to Key Public Data Resources</h1>
<h4 class="author">Morgan Feeney, Leighton Pritchard</h4>
<h4 class="date">2021 Presentation</h4>

</div>


<div id="summary">
<ul>
<li>This page lists some data resources that you may find useful while working on your capstone project.</li>
<li>While we have attempted to include common databases and resources you are likely to use, but it is <strong>not an exhaustive list of all the public databases</strong> that exist in the biomedical sciences.</li>
<li>Please send any suggestions for additional databases that should be included here, or report any broken links, to Dr. Morgan Feeney<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</li>
</ul>
</div>
<div id="multi-organism-databases" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Multi-organism databases</h1>
<ul>
<li><a href="https://biocyc.org/">BioCyc</a>
<ul>
<li>BioCyc integrates genome data with a comprehensive body of additional data including metabolic reconstructions, regulatory networks, protein features, orthologs, gene essentiality, and atom mappings.</li>
</ul></li>
<li><a href="https://www.ebi.ac.uk/biomodels/">BioModels</a>
<ul>
<li>BioModels is the repository of record for mathematical models of biological and biomedical systems. Many of the models in this database are manually validated and tested for reproducibility.</li>
</ul></li>
<li><a href="https://www.brenda-enzymes.org/">BRENDA</a>
<ul>
<li>BRENDA integrates functional and sequence data for enzymes.</li>
</ul></li>
<li><a href="https://www.ncbi.nlm.nih.gov/genbank/">Genbank</a>
<ul>
<li>GenBank is the NIH genetic sequence database, the reference of record for nucleotide sequences. Deposition of new sequence data (including genes, genomes, and metagenomes) into this database is usually a requirement for publication.</li>
</ul></li>
<li><a href="https://gold.jgi.doe.gov/">Genomes OnLine Database (GOLD)</a>
<ul>
<li>Comprehensive access to information regarding genome and metagenome sequencing projects, and their associated metadata, around the world.</li>
</ul></li>
<li><a href="https://www.ensembl.org/index.html">Ensembl Genome Browser</a>
<ul>
<li>A genome browser for vertebrate genomes that supports research in comparative genomics, evolution, sequence variation and transcriptional regulation.</li>
</ul></li>
<li><a href="https://ensemblgenomes.org/">Ensembl Genomes</a>
<ul>
<li>The equivalent of Ensembl Genome Browser for non-vertebrate genomes, including plants, bacteria, fungi and protists.</li>
</ul></li>
<li><a href="https://img.jgi.doe.gov/">IMG/M</a>
<ul>
<li>Supports the annotation, analysis and distribution of microbial genome and microbiome for those datasets sequenced at DOE’s Joint Genome Institute (JGI).</li>
</ul></li>
<li><a href="https://www.genome.jp/kegg/">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
<ul>
<li>A database resource that integrates pathway, sequence and enzyme function into high-level overviews of biological system, such as metabolic pathways, the cell, the organism and the ecosystem.</li>
</ul></li>
<li><a href="http://www.microbesonline.org/">microbesonline</a>
<ul>
<li>A comparative functional genomic database and workbench with tools for phylogenetic analysis and annotation, functional data storage/display/analysis, metabolic analysis and metafunctional genomics</li>
</ul></li>
<li><a href="https://www.ncbi.nlm.nih.gov/">NCBI Databases</a>
<ul>
<li>NCBI provides several databases (SRA, GEO, and GenBank are provided by NCBI but listed separately, here). In addition to raw sequence and transcriptome data, NCBI provides metadata describing biological samples, research projects, publications (PMIC) that are integrated and cross-referenced to each other.</li>
</ul></li>
<li><a href="http://pfam.xfam.org/">Pfam</a>
<ul>
<li>A large collection of protein domains and families, each represented by multiple sequence alignments and hidden Markov models (HMMs), that is a central reference for protein functional annotation.</li>
</ul></li>
<li><a href="https://www.rcsb.org/">RCSB Protein Data Bank (PDB)</a>
<ul>
<li>The repository of record for 3D structure data of large biological molecules (proteins, DNA, RNA, and complex assemblies). Deposition of structural data in this database is usually a requirement for publication.</li>
</ul></li>
<li><a href="https://rfam.xfam.org/">Rfam</a>
<ul>
<li>A resource that is analogous to Pfam for RNA sequences. A large collection of RNA families, each represented by multiple sequence alignments, consensus secondary structures and covariance models.</li>
</ul></li>
<li><a href="https://string-db.org/">STRING</a>
<ul>
<li>A database of known and predicted protein-protein interactions, and networks of those interactions.</li>
</ul></li>
<li><a href="https://www.ncbi.nlm.nih.gov/sra/">Sequence Read Archive (SRA)</a>
<ul>
<li>The repository of record for high throughput sequencing (sequence read) data. Deposition of read data in this database is usually a requirement for publication.</li>
</ul></li>
<li><a href="https://www.uniprot.org/">Uniprot</a>
<ul>
<li>A comprehensive, high-quality and freely accessible resource of protein sequence and functional information. It comprises four databases, the two key resources being Swiss-Prot - which is manually annotated and reviewed - and TrEMBL, which is automatically annotated and not reviewed.</li>
</ul></li>
</ul>
</div>
<div id="organism-specific-databases" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Organism-specific databases</h1>
<ul>
<li><a href="https://ecocyc.org/">EcoCyc</a>:
<ul>
<li>Database for the bacterium <em>Escherichia coli</em> K-12 MG1655</li>
</ul></li>
<li><a href="http://flybase.org/">FlyBase</a>:
<ul>
<li>Database of <em>Drosophila</em> Genes &amp; Genomes</li>
</ul></li>
<li><a href="http://www.informatics.jax.org/">Mouse Genome Informatics (MGI)</a>:
<ul>
<li>International database resource for the laboratory mouse, providing integrated genetic, genomic, and biological data to facilitate the study of human health and disease.</li>
</ul></li>
<li><a href="https://mycobrowser.epfl.ch/">MycoBrowser</a>:
<ul>
<li>A comprehensive genomic and proteomic data repository for pathogenic mycobacteria</li>
</ul></li>
<li><a href="https://www.pombase.org/">PomBase</a>:
<ul>
<li>Comprehensive database for the fission yeast <em>Schizosaccharomyces pombe</em>, providing structural and functional annotation, literature curation and access to large-scale data sets</li>
</ul></li>
<li><a href="https://pseudomonas.com/">pseudomonas.com</a>:
<ul>
<li>A repository for <em>Pseudomonas</em> genomes and functional annotations.</li>
</ul></li>
<li><a href="https://www.yeastgenome.org/">Saccharomyces Genome Database (SGD)</a>:
<ul>
<li>Comprehensive integrated biological information for the budding yeast <em>Saccharomyces cerevisiae</em> along with search and analysis tools to explore these data</li>
</ul></li>
<li><a href="http://strepdb.streptomyces.org.uk/">StrepDB</a>:
<ul>
<li>The <em>Streptomyces</em> Annotation Server</li>
</ul></li>
<li><a href="https://www.arabidopsis.org/">The Arabidopsis Information Resource (TAIR)</a>:
<ul>
<li>Genetic and molecular biology data for the model higher plant <em>Arabidopsis thaliana</em></li>
</ul></li>
<li><a href="https://wormbase.org/#012-34-5">WormBase</a>:
<ul>
<li>Accurate, current, accessible information concerning the genetics, genomics and biology of <em>C. elegans</em> and related nematodes</li>
</ul></li>
<li><a href="http://www.xenbase.org/entry/">XenBase</a>:
<ul>
<li>Resource that integrates all the diverse biological, genomic, genotype and phenotype data available from <em>Xenopus</em> research</li>
</ul></li>
<li><a href="http://zfin.org/">ZFin</a>:
<ul>
<li>Database of genetic and genomic data for the zebrafish (<em>Danio rerio</em>) as a model organism</li>
</ul></li>
</ul>
</div>
<div id="topic-specific-databases" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Topic-specific databases</h1>
<ul>
<li><a href="https://www.ebi.ac.uk/chembl/">ChEMBL</a>:
<ul>
<li>ChEMBL is a manually-curated database of bioactive molecules with drug-like properties that integrates chemical, bioactivity and genomic data.</li>
</ul></li>
<li><a href="http://www.kazusa.or.jp/codon/">Codon Usage Database</a>:
<ul>
<li>Database of codon usage tabulated from Genbank.</li>
</ul></li>
<li><a href="https://card.mcmaster.ca/">Comprehensive Antibiotic Resistance Database (CARD)</a>:
<ul>
<li>Bioinformatic database of resistance genes, their products and associated phenotypes.</li>
</ul></li>
<li><a href="https://metacyc.org/">MetaCyc</a>:
<ul>
<li>A curated database of experimentally elucidated metabolic pathways from all domains of life.</li>
</ul></li>
<li><a href="http://www.phi-base.org/">PHI-Base</a>
<ul>
<li>A curated database describing molecular and biological information on genes proven to affect the outcome of pathogen-host interactions.</li>
</ul></li>
<li><a href="http://rebase.neb.com/rebase/rebase.html">The Restriction Enzyme Database (REBASE)</a>:
<ul>
<li>A dynamic, curated database of restriction enzymes and related proteins</li>
</ul></li>
</ul>
</div>
<div id="other-data-resources" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Other data resources</h1>
<p>Some publications will deposit experimental or other supplementary data at generic data repositories that are not, strictly, databases. This is an increasingly common practice as funders insist on Open Science practices. Some examples of these resources are listed below.</p>
<ul>
<li><a href="https://figshare.com/">FigShare</a>
<ul>
<li>FigShare provides a repository for papers, FAIR data and non-traditional research outputs. It is able to assign DOIs to each deposited item.</li>
</ul></li>
<li><a href="https://github.com/">GitHub</a>
<ul>
<li>GitHub offers distributed version control and source code management, plus a range of features such as bug tracking, feature requests, task management, continuous integration and wikis. Most open source scientific software is available on GitHub. Many researchers use GitHub as a repository for supplementary data, with Zenodo as the DOI issuing authority.</li>
</ul></li>
<li><a href="https://osf.io/">osf.io</a>:
<ul>
<li>OSF is a free, open source project management tool that supports researchers throughout their entire project lifecycle. Many researchers use OSF as a long-term data repository, electronic lab notebook, or as the collaboration tool for their team’s research - this repository may then become supplementary information in a publication.</li>
</ul></li>
<li><a href="https://zenodo.org/">Zenodo</a>
<ul>
<li>Zenodo - developed at CERN as a catch-all repository for research data - provides a range of services for open science, including issuing of DOIs for research items. Many researchers use Zenodo to host preprints, archive public datasets, and software projects.</li>
</ul></li>
</ul>
<details>
<summary>
Click to see links to additional databases
</summary>
Looking for a database not listed here? Try looking in the <a href="https://www.oxfordjournals.org/nar/database/c">NAR database list</a> or the <a href="https://www.hsls.pitt.edu/obrc/">Online Bioinformatics Resources Collection (OBRC)</a>.
</details>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Or, if you’re feeling ambitious, make a pull request against the <a href="https://github.com/sipbs-compbiol/BM432">course material repository</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
